spec/builtins.test.sh

```diff
@@ -156,6 +156,13 @@ echo status=$?
 # OK bash stdout: status=2
 # OK dash stdout: status=127
 
+### Source with extra arguments
+lib=$TMP/to-be-sourced
+echo 'echo $@' > $lib
+. $lib foo bar  # dash doesn't have source
```

Comment:
This is a good test, but how about some more test cases?

What happens when you source a file inside a shell function?

How about testing multiple argv arrays?  If you need to make temp files you can do it with

```
tmp=$TMP/test-case-dir
mkdir -p $tmp
cd $tmp
echo lib > lib.sh
...

```

Reply:
Yeah, sourcing a file inside a shell function is the only way I know of to demonstrate the "locals are exposed" issue I mentioned in the comment. That would be a good one.

I'm not sure what you mean about multiple `argv` arrays. There's only one `argv` array per scope, right?